File: api/src/models/poBill.go
package models

import (
	"time"
)

type POBill struct {
	ID            uint       `gorm:"primaryKey" json:"id"`
	InvoiceNumber string     `gorm:"type:varchar(100);not null" json:"invoice_number"`
	VendorName    string     `gorm:"type:varchar(255);not null" json:"vendor_name"`
	Amount        float64    `gorm:"type:decimal(15,2);not null" json:"amount"`
	ReceivedDate  time.Time  `gorm:"not null" json:"received_date"`
	DueDate       time.Time  `gorm:"not null" json:"due_date"`
	Status        string     `gorm:"type:varchar(50);default:'pending';not null" json:"status"` // 'pending', 'paid'
	PaidDate      *time.Time `json:"paid_date,omitempty"`
	ReceiptImage  string     `gorm:"type:text" json:"receipt_image"`
	Notes         string     `gorm:"type:text" json:"notes"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

File: mobile/lib/features/po_bill/models/po_bill.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'po_bill.freezed.dart';
part 'po_bill.g.dart';

@freezed
class POBillModel with _$POBillModel {
  const factory POBillModel({
    int? id,
    @JsonKey(name: 'invoice_number') required String invoiceNumber,
    @JsonKey(name: 'vendor_name') required String vendorName,
    required double amount,
    @JsonKey(name: 'received_date') required DateTime receivedDate,
    @JsonKey(name: 'due_date') required DateTime dueDate,
    @Default('pending') String status, // 'pending' or 'paid'
    @JsonKey(name: 'paid_date') DateTime? paidDate,
    @JsonKey(name: 'receipt_image') @Default('') String receiptImage,
    @Default('') String notes,
    @JsonKey(name: 'created_at') DateTime? createdAt,
    @JsonKey(name: 'updated_at') DateTime? updatedAt,
  }) = _POBillModel;

  factory POBillModel.fromJson(Map<String, dynamic> json) => _$POBillModelFromJson(json);
}
